modules/pxp-module-puppet

```diff
@@ -282,6 +282,10 @@ def action_run(input)
         1
       end
 
+      # flush the stdout/stderr before writing the exitcode
+      # file to avoid pxp-agent reading incomplete output
+      $stdout.fsync
```

Comment:
After a second thought, I actually don't agree, since:

1. we can't guarantee valid JSON on stdout in the rescue block anyway - as by the time we print the response, there might have been something already written to the stdout
2. the print in the rescue block is followed by exit which flushes the buffers implicitly